src/blob/index/bptree/node.rs

```diff
@@ -0,0 +1,103 @@
+use super::prelude::*;
+
+#[derive(Debug, PartialEq)]
+pub(super) struct Node {
+    keys: Vec<Vec<u8>>,
+    offsets: Vec<u64>,
+}
+
+impl Node {
+    pub(super) fn new(keys: Vec<Vec<u8>>, offsets: Vec<u64>) -> Self {
+        assert!(keys.len() != 0);
```

Comment:
I think it's a not good practice to use assertions in working code

Reply:
And btw, in that case you should use `!keys.is_empty()` (:

Reply:
I mentioned this practice in std lib: for example, vector panics, if size is more than `isize::MAX`.

B+ Tree doesn't assume node to be without keys and children pointers amount has to be equal to `keys.len() + 1`. These are invariants, which can't be violated (So this state must be impossible). 
I don't violate them, so assert will never panic. But if we'll modify code, we may forget about it, so I decided to do it in a such way. But what are your suggestions?